src/plugins_types.h

```diff
@@ -77,19 +77,21 @@ struct lysp_module;
  * required format via ::ly_type_print_clb implementation.
  */
 
-/**
- * @brief Helper function for various plugin functions to generate error information structure.
- *
- * @param[in] level Error level of the error.
- * @param[in] code Code of the error.
- * @param[in] vecode Validity error code in case of LY_EVALID error code.
- * @param[in] msg Error message.
- * @param[in] path Path to the node causing the error.
- * @param[in] apptag Error-app-tag value.
+/*
+ * @brief Create and fill error structure.
+ *
+ * Helper function for various plugin functions to generate error information structure.
+ *
+ * @param [in] ecode Code of the error to fill. If there is not enough memory for allocating error record then this code is
+ * changed to LY_EMEM, but NULL is returned.
+ * @param [in] vecode Validity error code in case of LY_EVALID error code.
+ * @param [in] path Path to the node causing the error.
+ * @param [in] apptag Error-app-tag value.
+ * @param [in] err_msg error formating message.
  * @return NULL in case of memory allocation failure.
- * @return Created error information structure that can be freed using ::ly_err_free().
+ * @return Created error information structure that can be freed using ::ly_err_free() or passed back from callback into libyang.
  */
-struct ly_err_item *ly_err_new(LY_LOG_LEVEL level, LY_ERR code, LY_VECODE vecode, char *msg, char *path, char *apptag);
+struct ly_err_item *ly_err_new(LY_ERR *ecode, LY_VECODE vecode, char *path, char *apptag, const char *err_msg, ...);
```

Comment:
I suggest the same simplification as for the API function.

Reply:
which API function? I'm confused, this is the header making ly_err_new() public. The function is implemented in log.c just because it extends the functionality of some other functions present there, but the only expected use is for type plugins, so the declaration is in this header.

Reply:
Right, sorry, I somehow missed the fact that this is just a prototype.